llvm-runtimes/libgcc/libgcc-19.1.4.ebuild

```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit crossdev llvm.org toolchain-funcs
+
+DESCRIPTION="Compiler runtime library for clang, compatible with libgcc_s"
+HOMEPAGE="https://llvm.org/"
+
+LICENSE="Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~mips ~ppc64 ~riscv ~x86 ~amd64-linux ~arm64-macos ~ppc-macos ~x64-macos"
+
+DEPEND="
+	llvm-runtimes/compiler-rt:${LLVM_MAJOR}
+	=llvm-runtimes/libunwind-${PV}
+	!!sys-devel/gcc
+"
+BDEPEND="
+	llvm-core/clang:${LLVM_MAJOR}
+"
+
+LLVM_COMPONENTS=( llvm-libgcc )
+llvm.org_set_globals
+
+src_compile() {
+	$(tc-getCC) -E -xc \
+		"${WORKDIR}/llvm-libgcc/gcc_s.ver.in" \
+		-o gcc_s.ver || die
+	$(tc-getCC) -nostdlib \
+		-Wl,--version-script,gcc_s.ver \
+		-Wl,--whole-archive \
+		"${EPREFIX}/usr/lib/clang/${LLVM_MAJOR}/lib/linux/libclang_rt.builtins-${CTARGET%%-*}.a" \
+		"${EPREFIX}/usr/lib/libunwind.a" \
```

Comment:
```suggestion
		"${ESYSROOT}/usr/$(get_libdir)/libunwind.a" \
```

Reply:
Actually, any reason not to used the shared library here?

Reply:
When I'm trying to use the shared library, I'm getting

```
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_DeleteException' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_ForcedUnwind' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetDataRelBase' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetLanguageSpecificData' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetRegionStart' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetTextRelBase' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_RaiseException' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_Resume' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.3' to symbol '_Unwind_GetCFA' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.3' to symbol '_Unwind_Resume_or_Rethrow' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_Find_FDE' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetGR' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_GetIP' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_SetGR' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '_Unwind_SetIP' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.3' to symbol '_Unwind_Backtrace' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.3' to symbol '_Unwind_FindEnclosingFunction' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_4.2.0' to symbol '_Unwind_GetIPInfo' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '__deregister_frame' failed: symbol not defined
ld.lld: error: version script assignment of 'GCC_3.0' to symbol '__register_frame' failed: symbol not defined
```